src/mathMLTree.js

```diff
@@ -116,34 +109,114 @@ export class MathNode {
  */
 export class TextNode {
     text: string;
+    needsEscape: boolean;
 
-    constructor(text: string) {
+    constructor(text: string, needsEscape: boolean = true) {
         this.text = text;
+        this.needsEscape = needsEscape;
     }
 
     /**
      * Converts the text node into a DOM text node.
      */
     toNode(): Node {
-        return document.createTextNode(this.text);
+        return document.createTextNode(this.toText());
+    }
+
+    /**
+     * Converts the text node into escaped HTML markup
+     * (representing the text itself).
+     */
+    toMarkup(): string {
+        return this.toText();
+    }
+
+    /**
+     * Converts the text node into an escaped string
+     * (representing the text iteself).
+     */
+    toText(): string {
+        return this.needsEscape ? utils.escape(this.text) : this.text;
+    }
+}
+
+/**
+ * This node represents a space, but may render as <mspace.../> or as text,
+ * depending on the width.
+ */
+class SpaceNode {
+    width: number;
+    character: ?string;
+
+    /**
+     * Create a Space node with width given in CSS ems.
+     */
+    constructor(width: number) {
+        this.width = width;
+        // See https://www.w3.org/TR/2000/WD-MathML2-20000328/chapter6.html
+        // for a table of space-like characters.  We consistently use the
+        // &LongNames; because Unicode does not have single characters for
+        // &ThickSpace; (\u2005\u200a) and all negative spaces.
```

Comment:
I don't don't understand this comment.  I don't see `$LongNames;` anywhere.

Reply:
I meant `&ThickSpace;` is an example of a long name between a `&` and `;`, as opposed to the Unicode equivalent.  See the table in Section 6.1.4.  I could change to `&CharacterNames;` or just "character names" if that would be clearer.